src/com/xilinx/rapidwright/rwroute/RoutableNode.java

```diff
@@ -0,0 +1,570 @@
+	public int getIndex() {
+		return this.index;
+	}
+	
+	@Override
+	public boolean isInConnectionBoundingBox(Connection connection) {		
+		return this.endTileXCoordinate > connection.getXMinBB() && this.endTileXCoordinate < connection.getXMaxBB() && this.endTileYCoordinate > connection.getYMinBB() && this.endTileYCoordinate < connection.getYMaxBB();
+	}
+	
+	@Override
+	public Node getNode() {
+		return this.node;
+	}
+
+	@Override
+	public boolean isTarget() {
+		return this.target;
+	}
+
+	@Override
+	public void setTarget(boolean isTarget) {
+		this.target = isTarget;	
+	}
+
+	@Override
+	public RoutableType getRoutableType() {
+		return this.type;
+	}
+
+	@Override
+	public float getDelay() {
+		return this.delay;
+	}
+	@Override
+	public short getEndTileXCoordinate() {
+		return this.endTileXCoordinate;
+	}
+
+	@Override
+	public void setEndTileXCoordinate(short endTileXCoordinate) {
+		this.endTileXCoordinate = endTileXCoordinate;
+	}
+
+	@Override
+	public void setEndTileYCoordinate(short endTileYCoordinate) {
+		this.endTileYCoordinate = endTileYCoordinate;
+	}
+
+	@Override
+	public short getEndTileYCoordinate() {
+		return this.endTileYCoordinate;
+	}
+	
+	@Override
+	public float getBaseCost() {
+		return this.baseCost;
+	}
+
+	public boolean childrenNotSet() {
+		return this.children == null;
+	}
+
+	@Override
+	public List<Routable> getChildren() {
+		return this.children;
+	}
+
+	@Override
+	public void setDelay(short delay) {
+		this.delay = delay;
+	}
+
+	@Override
+	public int manhattanDistToSink(Routable sink) {
+		return Math.abs(this.getEndTileXCoordinate() - sink.getEndTileXCoordinate()) + Math.abs(this.getEndTileYCoordinate() - sink.getEndTileYCoordinate());
+	}
+
+	@Override
+	public void setRoutableType(RoutableType type) {
+		this.type = type;
+	}
+
+	@Override
+	public short getLength() {
+		return this.length;
+	}
+	
+	@Override
+	public void setLowerBoundTotalPathCost(float totalPathCost) {
+		this.lowerBoundTotalPathCost = totalPathCost;
+		this.setVisited(true);
+	}
+	
+	@Override
+	public void setUpstreamPathCost(float newPartialPathCost) {
+		this.upstreamPathCost = newPartialPathCost;
+	}
+	
+	@Override
+	public float getLowerBoundTotalPathCost() {
+		return this.lowerBoundTotalPathCost;
+	}
+	
+	@Override
+	public float getUpstreamPathCost() {
+		return this.upstreamPathCost;
+	}
+
+	@Override
+	public Map<SitePinInst, Integer> getUsersConnectionCounts() {
+		return usersConnectionCounts;
+	}
+	
+	@Override
+	public void addUser(SitePinInst source) {
+		if(this.usersConnectionCounts == null) {
+			this.usersConnectionCounts = new HashMap<>();
+		}
+		Integer users = this.usersConnectionCounts.getOrDefault(source, 0);
+		this.usersConnectionCounts.put(source, users + 1);
+	}
+	
+	@Override
+	public int numUniqueUsers() {
+		if(this.usersConnectionCounts == null) {
+			return 0;
+		}
+		return this.usersConnectionCounts.size();
+	}
+	
+	@Override
+	public void reduceConnectionCountOfUser(SitePinInst source) {
```

Comment:
Since it looks like this method is the opposite of `addUser`, could their names be made consistent?

Reply:
True. addUser changed to increaseConnectionCountOfUser.